src/temporal/t.rast.boxplot/t.rast.boxplot.py

```diff
@@ -1,631 +1,802 @@
+        else:
+            date_fmt = mdates.DateFormatter("%Y-%m-%d")
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+    if "hour" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("hour %H")
+        start_time = mdates.date2num(rast_dates[0]) - 1 / 48 * temp_lngt
+        end_time = mdates.date2num(rast_dates[-1]) + 1 / 48 * temp_lngt
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+            ax.set_xlim(mdates.num2date(start_time), mdates.num2date(end_time))
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+            ax.set_ylim(mdates.num2date(start_time), mdates.num2date(end_time))
+    if "minute" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("min. %M")
+        start_time = mdates.date2num(rast_dates[0]) - 1 / 2880 * temp_lngt
+        end_time = mdates.date2num(rast_dates[-1]) + 1 / 2880 * temp_lngt
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+            ax.set_xlim(mdates.num2date(start_time), mdates.num2date(end_time))
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+            ax.set_ylim(mdates.num2date(start_time), mdates.num2date(end_time))
+    if "second" in temp_unit:
+        if bool(date_format):
+            date_fmt = mdates.DateFormatter(date_format)
+        else:
+            date_fmt = mdates.DateFormatter("sec. %S")
+        start_time = mdates.date2num(rast_dates[0]) - 1 / 172800 * temp_lngt
+        end_time = mdates.date2num(rast_dates[-1]) + 1 / 172800 * temp_lngt
+        if vertical:
+            ax.xaxis.set_major_formatter(date_fmt)
+            ax.set_xlim(mdates.num2date(start_time), mdates.num2date(end_time))
+        else:
+            ax.yaxis.set_major_formatter(date_fmt)
+            ax.set_ylim(mdates.num2date(start_time), mdates.num2date(end_time))
+    return ax
+
+
+def main(options, flags):
+    """
+    Draws the boxplot of raster values. Optionally, this is done per
+    category of a zonal raster layer
+    """
+
+    global matplotlib
+    global plt
+
+    # lazy import matplotlib
+    try:
+        import matplotlib
+
+        matplotlib.use("WXAgg")
+        from matplotlib import pyplot as plt
+    except ModuleNotFoundError:
+        gs.fatal(_("matplotlib is not installed"))
+
+    # Plot format options
+    fontsize = options["fontsize"]
+    if fontsize:
+        int(fontsize)
+        plt.rcParams["font.size"] = fontsize
+    else:
+        fontsize = 10
+    if options["rotate_labels"]:
+        rotate_label = float(options["rotate_labels"])
+    grid = flags["g"]
+
+    # Get range (if defined)
+    whisker_range = get_range(options["range"])
```

Comment:
Simplified:

```python
whisker_range = float(options["range"])
range_min_val = 0
if whisker_range <= range_min_val:
    gs.fatal(
        _(
            "The range value need to be larger than {min_val}".format(
                min_val=range_min_val,
            )
        )
    )
```

`get_range()` func is no needed, range option has default value (answer: 1.5, [line 69](https://github.com/OSGeo/grass-addons/pull/781/commits/bdcb338bac3bd1d81a5dd093ea6925d14bcf88c1#diff-97f9248c2483025a0e4d5b85171647b08c946e5cbca74b9e3db24ebaf4d3dc15R69)).